src/cloud-providers/util.go

```diff
@@ -102,6 +114,21 @@ func GetBestFitInstanceType(sortedInstanceTypeSpecList []InstanceTypeSpec, vcpus
 
 }
 
+// Implement the GetBestFitInstanceTypeWithGPU function
+func GetBestFitInstanceTypeWithGPU(sortedInstanceTypeSpecList []InstanceTypeSpec, gpus, vcpus, memory int64) (string, error) {
```

Comment:
Should the original GetBestFitInstanceType make sure 0 GPUs instance type are returned (in case gpuCount is 0)?